woodwork/config.py

```diff
@@ -1,6 +1,7 @@
 CONFIG_DEFAULTS = {
     'categorical_threshold': 10,
     'numeric_categorical_threshold': -1,
+    'email_inference_regex': r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)",
```

Comment:
We have a documentation guide that explains the various Woodwork configuration options. I think we should also add a brief description to that guide that mentions this option: https://woodwork.alteryx.com/en/stable/guides/setting_config_options.html#Available-Config-Settings

You can do that by updating the `setting_config_options.ipynb` file in the `docs/source/guides` folder.